drivers/storage/portworx/status_test.go

```diff
@@ -177,3 +184,283 @@ func TestSetupContextWithToken(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateStorageNodePhase(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// Create the mock servers that can be used to mock SDK calls
+	mockClusterServer := mock.NewMockOpenStorageClusterServer(mockCtrl)
+	mockNodeServer := mock.NewMockOpenStorageNodeServer(mockCtrl)
+
+	// Start a sdk server that implements the mock servers
+	sdkServerIP := "127.0.0.1"
+	sdkServerPort := 21883
+	mockSdk := mock.NewSdkServer(mock.SdkServers{
+		Cluster: mockClusterServer,
+		Node:    mockNodeServer,
+	})
+	mockSdk.StartOnAddress(sdkServerIP, strconv.Itoa(sdkServerPort))
+	defer mockSdk.Stop()
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Status: corev1.StorageClusterStatus{
+			Phase: "Initializing",
+		},
+	}
+
+	testCases := []struct {
+		name          string
+		pxNode        *api.StorageNode
+		storageNode   *corev1.StorageNode
+		expectedPhase string
+	}{
+		{
+			name: "with-no-conditions",
+			storageNode: &corev1.StorageNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "k8s-node",
+					Namespace: cluster.Namespace,
+				},
+				Status: corev1.NodeStatus{},
+			},
+			expectedPhase: "Initializing",
+		},
+		{
+			name: "with-only-succeeded-init-condition",
+			storageNode: &corev1.StorageNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "k8s-node",
+					Namespace: cluster.Namespace,
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:               corev1.NodeInitCondition,
+							Status:             corev1.NodeSucceededStatus,
+							LastTransitionTime: metav1.NewTime(time.Now()),
+						},
+					},
+				},
+			},
+			expectedPhase: "Initializing",
+		},
+		{
+			name: "with-succeeded-init-condition-and-state-condition-without-status",
+			storageNode: &corev1.StorageNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "k8s-node",
+					Namespace: cluster.Namespace,
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:               corev1.NodeInitCondition,
+							Status:             corev1.NodeSucceededStatus,
+							LastTransitionTime: metav1.NewTime(time.Now()),
+						},
+						{
+							Type:               corev1.NodeStateCondition,
+							LastTransitionTime: metav1.NewTime(time.Now().Add(time.Minute)),
+						},
+					},
+				},
+			},
+			expectedPhase: "Initializing",
+		},
+		{
+			name: "with-old-succeeded-init-condition-and-valid-state-condition",
+			pxNode: &api.StorageNode{
+				Id:                "id",
+				SchedulerNodeName: "k8s-node",
+				Status:            api.Status_STATUS_NOT_IN_QUORUM,
```

Comment:
It might be a naive question, but why we don't need to test other statuses? such as Status_STATUS_MAINTENANCE.

Reply:
There is no reason as such. Just wanted to take something other than ONLINE/OK because that is what we end up checking in most places.
Here we wanted the check that NodeState is reflected in the final Phase, instead of NodeInit. So it does not matter what is the Portworx node status is. Anything other than Initializing would have done the job.

Reply:
Thanks for the explaination.